docker/cytoscape/tests/connect_cytoscape.py

```diff
@@ -0,0 +1,242 @@
+            sys.stdout.flush()
+        r = requests.delete(url = URL)
+        verbose_=checkresponse(r, verbose=verbose)
+        if (verbose) or (verbose_):
+            print("'"+URL+"'")
+            sys.stdout.flush()
+        res=r.json()
+        if len(res["errors"]) > 0:
+            raise ValueError(res["errors"][0])
+
+    elif (method == "POST") or (method == "P"):
+        if verbose:
+            print("'"+baseurl+"'")
+            sys.stdout.flush()
+        r = requests.post(url = baseurl, json = PARAMS)
+        verbose_=checkresponse(r, verbose=verbose)
+        if (verbose) or (verbose_):
+            verbose=True
+            print(r.content)
+            sys.stdout.flush()
+
+        res=r.json()
+        if "errors" in res.keys():
+            if len(res["errors"]) > 0:
+                raise ValueError(res["errors"][0])
+        if not verbose:
+            if "data" in res.keys():
+                res=res["data"]
+        else:
+            res=verbose_
+
+    elif (method == "PUT"):
+        if verbose:
+            print("'"+baseurl+"'")
+            sys.stdout.flush()
+        r = requests.put(url = baseurl, json = body)
+
+        verbose_=checkresponse(r, verbose=verbose)
+        if (verbose) or (verbose_):
+            verbose=True
+            print(r.content)
+            sys.stdout.flush()
+
+        try:
+            res=r.json()
+            if "errors" in res.keys():
+                if len(res["errors"]) > 0:
+                    raise ValueError(res["errors"][0])
+        except JSONDecodeError:
+            if not r.text:
+                res = {}
+            else:
+                raise
+
+    elif (method=="HTML") or (method == "H") or (method=="HELP"):
+        URL = baseurl
+        if verbose:
+            print("'"+URL+"'")
+            sys.stdout.flush()
+        response = requests.get(URL, params=PARAMS)
+
+        res = response.text.split("\n")
+        def clean(x):
+            r=x.split("</p>")[0].split(">")[-1]
+            return r
+        res="\n".join([ clean(x) for x in res ])
+
+    res=handle_status_codes(res,verbose=verbose)
+
+    return res
+
+def _status(cyrest_client, host, port, verbose=False):
+    """
+    Checks the status of your CyREST server.
+    """
+    _url = 'http://' + host + ':' + str(port) + '/' + 'v1' + '/'
+    try:
+        response=api(url=_url, method="GET", verbose=verbose)
+        print('CyREST online!')
+        return response
+    except Exception as e:
+        print('Connecting to Cytoscape ...')
+
+##########################################################################################################
+##########################################################################################################
+       
+
+def check_cyrest(port):
+    print('start supervisord client')
+    try:
+        
+        subprocess.run([
+            "/usr/bin/supervisord", "-c", "/etc/supervisor/conf.d/supervisord.conf"
+        ],
```

Comment:
When does this run?  I looked like the Dockerfile command already started supervisord.

Reply:
This is part of a test script that connects to cytoscape within the container. You can run this script in interactive mode by calling `bash tests/connect.sh` from the root directory of the container.